File: backend/src/main/java/com/project/backend/domain/book/dto/BookDto.java
package com.project.backend.domain.book.dto;

import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;

@Getter
@Setter
public class BookDto {

    @NonNull
    private int id;

    @NonNull
    private String title;

    @NonNull
    private String author;

    @NonNull
    private String description;

    @NonNull
    private String image;

    @NonNull
    private int favoriteCount;
}

File: backend/src/main/java/com/project/backend/domain/book/dto/BookDTO.kt
package com.project.backend.domain.book.dto

/**
 * -- Book엔티티의 DTO --
 *
 * @author -- 정재익 --
 * @since -- 1월 27일 --
 */
data class BookDTO(
    val id: Long?,
    val title: String = "제목 정보 없음",
    val author: String = "작가 정보 없음",
    val description: String = "설명 정보 없음",
    val image: String = "이미지 정보 없음",
    val isbn: String,
    val ranking: Int?,
    var favoriteCount: Int? = 0)
